gwswlib/hydx.py

```diff
@@ -8,6 +8,47 @@ class Hydx:
     def __init__(self):
         self.connection_nodes = []
 
+    def check_import_data(self):
+        self._check_on_unique(
+            self.connection_nodes, "identificatieknooppuntofverbinding", True, "knoop"
+        )
+
+    def _check_on_unique(
+        self,
+        records,
+        unique_field,
+        remove_doubles=False,
+        item_name_for_logging="",
+        log_level=logging.WARNING,
+    ):
+
+        values = [m.__dict__[unique_field] for m in records]
+
+        if len(set(values)) == len(values):
+            return True, set()
+
+        doubles = []
+        value_set = set()
+
+        for i in reversed(range(0, len(records))):
```

Comment:
Loopen met `range()` is vaak niet nodig. Normaliter kan je "gewoon" door de lijst heen lopen.

Misschien gaat het trouwens ook fout omdat je records tijdens de loop verwijdert: gaat het daarna goed met het ophalen van de resterende items?
Alternatief is een lijstje bijhouden van te verwijderen items en dan aan het eind (buiten de loop) ze in één keer te verwijderen?

Maar.... daarvoor heb je wel die `i` weer nodig. Ken je `enumerate()`? Die doet denk ik precies wat je wilt:

    lijstje = [10, 20, 30, 40]
    for i, item in enumerate(lijstje):
        # i is index, het ding dat je nu via range() doet.